### PR TITLE
[ST] Add new fields to testStorage

### PR DESCRIPTION
**Pull request by jankalinic**
### Type of change

- Enhancement / new feature
- Refactoring

### Description

This pull request introduces a series of new fields to the testStorage class. The primary objective is to streamline and simplify access to test related data. These newly added fields have their primary usage within the context of MM STs.

Previously, these variables were defined separately for each individual test case, resulting in redundant variables from created from testStorage. With this enhancement, PR aims to centralize the management of these test-related variables within the testStorage class, aligning them with other similar variables and improving overall code organization and maintainability.

Also worth mentioning that a big part of this PR is refactoring of classes that use these newly added variables.

### Checklist
- [ ] Make sure all tests pass
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally

### CODE REVIEW

**Comment by jankalinic** [comment]
@strimzi-ci  run tests --cluster-type=ocp --cluster-version=4.14 --install-type=bundle --testcase=MirrorMaker2ST

**Comment by jankalinic** [comment]
@strimzi-ci run tests --cluster-type=ocp --cluster-version=4.14 --install-type=bundle --testcase=MirrorMaker2ST

File: systemtest/src/main/java/io/strimzi/systemtest/storage/TestStorage.java
Comment (henryZrncik) [inline systemtest/src/main/java/io/strimzi/systemtest/storage/TestStorage.java:82]: looking on how mm2 test are refactored i think this can be a bit tricky, it is refactored in such a manner that we are producing something to source cluster topic called either `testStorage.getTopicName` or `testStorage.getSourceTopicName` and latter expect it to be  mirrored to the topic called `testStorage.getTargetTopicName` ? 
the logic I am missing here is maping like `kafkaClusterSourceName + "." + testStorage.getTopicName();` when trying to obtain name of mirrored topic in target cluster.

**Review by henryZrncik — APPROVED** [review]
LGTM except one thing I am probably missing.

File: systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMaker2ST.java
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMaker2ST.java:159]: This will not work FMPOV.
The name of the mirrored topic in target cluster follows this pattern - `<source_cluster_name>.<topic_name>` and you are using here `<topic_name>-source`.

Yeah there is one or two testcases were we are using just source topic name, as we have a special configuration there (the `IdentityReplicationPolicy`), but this is not the case for all other testcases.

File: systemtest/src/main/java/io/strimzi/systemtest/storage/TestStorage.java
Comment (jankalinic) [reply systemtest/src/main/java/io/strimzi/systemtest/storage/TestStorage.java:82]: You are absolutely right! I confused the logic there. Changing that now.

**Comment by jankalinic** [comment]
@strimzi-ci run tests --cluster-type=ocp --cluster-version=4.14 --testcase=MirrorMaker2ST,LoggingChangeST,MirrorMakerST,SecurityST,OauthPlainST,OpenTelemetryST,RackAwarenessST,OauthPasswordGrantsST

**Comment by jankalinic** [comment]
@strimzi-ci run tests --cluster-type=ocp --cluster-version=4.14 --testcase=MirrorMaker2ST,LoggingChangeST,MirrorMakerST,SecurityST,OauthPlainST,OpenTelemetryST,RackAwarenessST,OauthPasswordGrantsST

File: systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java
Comment (see-quick) [inline systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java:94]: I think you can access the whole `LabelSelector` instance from TestStorage and you do not need to again call `KafkaResource.getLabelSelector...`.

File: systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java
Comment (see-quick) [inline systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java:95]: Same for this one, and I don't know how many cases are in other modifications but apply this in all of them thanks.

File: systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java
Comment (see-quick) [inline systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java:1088]: I don't know if we have KMM2 label selector in TestStorage, but if not we can simply add and use it as we have Kafka, ZK selectors there.

**Review by see-quick — APPROVED** [review]
LGTM, just a few nits, which need to be addressed. Anyway, good job! 👍

File: systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java
Comment (jankalinic) [reply systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java:94]: Good tip! Thanks, this also seems to apply to zookeeper label selector.

**Comment by jankalinic** [comment]
@strimzi-ci run tests --cluster-type=ocp --cluster-version=4.14 --install-type=bundle --testcase=io.strimzi.systemtest.security.oauth.OauthPasswordGrantsST#testPasswordGrantsKafkaMirrorMaker2

File: systemtest/src/test/resources/junit-platform.properties
Comment (im-konge) [inline systemtest/src/test/resources/junit-platform.properties:4]: these shouldn't be changed, you should use variables for that or revert the changes before commit

**Comment by scholzj** [comment]
This is not ready for merge because it does not have the required number of reviews from maintainers.

**Review by im-konge — APPROVED** [review]

**Review by Frawless — APPROVED** [review]

**Pull request merged**
